### PR TITLE
[Api] Adding product to cart with proper product price per channel

### PR DESCRIPTION
**Pull request by Tomanhez**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: features/cart/shopping_cart/adding_product_to_cart_with_prices_dependent_on_channel.feature [hunk 1]
```diff
@@ -21,6 +21,13 @@ Feature: Adding a product to cart with prices dependent on a channel
         When I add product "Leprechaun's Gold" to the cart
         Then I should see "Leprechaun's Gold" with unit price "€10.00" in my cart
 
+    @api
+    Scenario: Buying a product in default currency for browsed channel
+        Given I change my current channel to "Web-EU"
+        When I pick up my cart
+        And I add product "Leprechaun's Gold" to the cart
+        Then I should see "Leprechaun's Gold" with unit price "€10.00" in my cart
+
```
Comment (GSadee) [inline features/cart/shopping_cart/adding_product_to_cart_with_prices_dependent_on_channel.feature:30]: There is no need to add this scenario, you could add an `@api` tag to the previous scenario

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php [hunk 2]
```diff
@@ -334,6 +334,24 @@ public function iShouldBeNotifiedThatQuantityOfAddedProductCannotBeLowerThan1():
...
+    /**
+     * @Then /^I should see "([^"]+)" with unit price ("[^"]+") in my cart$/
+     */
+    public function iShouldSeeProductWithUnitPriceInMyCart(string $productName, int $unitPrice): void
+    {
+        $response = $this->cartsClient->getLastResponse();
+
+        foreach ($this->responseChecker->getValue($response, 'items') as $item) {
+            if ($item['productName'] === $productName) {
+                Assert::same($item['total'], $unitPrice);
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CartContext.php:346]:
```suggestion
                Assert::same($item['unitPrice'], $unitPrice);
```
?

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php [hunk 3]
```diff
@@ -334,6 +334,24 @@ public function iShouldBeNotifiedThatQuantityOfAddedProductCannotBeLowerThan1():
...
+
+                return;
+            }
+        }
+
+        throw new \InvalidArgumentException(sprintf('Product with %s does not exist', $productName));
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CartContext.php:352]:
```suggestion
        throw new \InvalidArgumentException(sprintf('The product %s does not exist', $productName));
```

File: features/cart/shopping_cart/adding_product_to_cart_with_prices_dependent_on_channel.feature (hunk 1 above)
Comment (Tomanhez) [reply features/cart/shopping_cart/adding_product_to_cart_with_prices_dependent_on_channel.feature:30]: Ok but it doesn't work without `When I pick up my cart`

Comment (AdamKasp) [reply features/cart/shopping_cart/adding_product_to_cart_with_prices_dependent_on_channel.feature:30]: you can pick up a cart in step `I add product "Leprechaun's Gold" to the cart` if cart does not exist :)

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php [hunk 4]
```diff
@@ -115,6 +115,10 @@ public function theAdministratorTryToSeeTheSummaryOfCart(?string $tokenValue): v
      */
     public function iAddThisProductToTheCart(ProductInterface $product, ?string $tokenValue): void
     {
+        if (!$this->sharedStorage->has('cart_token')) {
+            $this->pickupCart();
+        }
+
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CartContext.php:121]: Should be removed, this logic is already in `putProductToCart()` method if I see correctly, and what is more, you have `$tokenValue` which is a value taken from `sharedStorage`

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/OrderItem.xml [hunk 5]
```diff
@@ -48,7 +48,7 @@
         </attribute>
         <attribute name="unitPrice">
             <group>admin:order_item:read</group>
-            <group>shop:order_item:read</group>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serialization/OrderItem.xml:51]: Why did you remove that?

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/OrderItem.xml [hunk 6]
```diff
@@ -89,7 +89,6 @@
         <attribute name="total">
             <group>admin:order:read</group>
             <group>admin:order_item:read</group>
-            <group>shop:order_item:read</group>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serialization/OrderItem.xml:92]: Why did you remove that?

**Review by arti0090 — APPROVED** [review]

**Review by AdamKasp — APPROVED** [review]

**Comment by AdamKasp** [comment]
Thank you, Tomasz! :1st_place_medal:
